speculos/mcu/seproxyhal.py

```diff
@@ -48,23 +48,61 @@ class SephTag(IntEnum):
 RENDER_METHOD = RenderMethods(0, 1)
 
 
-def ticker(add_tick):
-    """
-    Function ran in a thread to trigger a tick at a periodic interval.
+class TimeTickerDaemon(threading.Thread):
+    def __init__(self, add_tick, *args, **kwargs):
+        super(TimeTickerDaemon, self).__init__(name="time_ticker", daemon=True)
+        self.stop: bool = False
+        self.pause: bool = False
+        self.resume: threading.Condition = threading.Condition()
+        self.add_tick: Callable = add_tick
 
-    There is no need to prevent clock drift.
-    """
+        """
+        Initializes the Ticker Daemon
+        The goal of this daemon is to emulate the time spent in the application by sending it
+        ticker events at a regular interval TICKER_DELAY.
+        This daemon can be paused and resumed through it's API to stop the flow of time in the MCU
+
+        :param add_tick: The callback function to add a Ticker Event to the packet manager
+        :type add_tick: Backend
+        """
+
+    def pause_emulated_time(self):
+        """
+        Pause time emulation done by the daemon, no ticker event will be sent until resume
+        """
+        self.pause = True
+
+    def resume_emulated_time(self):
+        """
+        Resume time emulation done by the daemon, no ticker event will be sent until resume
+        """
+        self.pause = False
+        with self.resume:
+            self.resume.notify()
+
+    def _wait_if_time_paused(self):
+        """
+        Internal function to handle the pause
+        """
+        while self.pause:
+            with self.resume:
+                self.resume.wait()
 
-    while True:
-        add_tick()
-        time.sleep(TICKER_DELAY)
+    def run(self):
+        """
+        Main thread function
+        """
+        while not self.stop:
```

Comment:
I don't see where this property is changed.
I see that the `PacketDaemon` class changes its own `.stop` property, but not this `TimeTickerDaemon` class. It's initialized at `False` and that's it?

Reply:
Removed